include/exadg/functions_and_boundary_conditions/evaluate_functions.h

```diff
@@ -203,21 +224,137 @@ struct FunctionEvaluator<1, dim, Number>
 
     for(unsigned int d = 0; d < dim; ++d)
     {
-      Number array[dealii::VectorizedArray<Number>::size()];
       for(unsigned int v = 0; v < dealii::VectorizedArray<Number>::size(); ++v)
       {
         dealii::Point<dim>     q_point;
         dealii::Tensor<1, dim> normal;
-        for(unsigned int d = 0; d < dim; ++d)
+        for(unsigned int i = 0; i < dim; ++i)
         {
-          q_point[d] = q_points[d][v];
-          normal[d]  = normals[d][v];
+          q_point[i] = q_points[i][v];
+          normal[i]  = normals[i][v];
         }
         function_with_normal->set_time(time);
         function_with_normal->set_normal_vector(normal);
-        array[v] = function_with_normal->value(q_point, d);
+        value[d][v] = function_with_normal->value(q_point, d);
+      }
+    }
+
+    return value;
+  }
+};
+
+template<int dim, typename Number>
+struct FunctionEvaluator<2, dim, Number>
+{
+  static inline DEAL_II_ALWAYS_INLINE //
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
+    value(std::shared_ptr<dealii::Function<dim>>                      function,
+          dealii::Point<dim, dealii::VectorizedArray<Number>> const & q_points,
+          double const &                                              time)
+  {
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> value;
+
+    for(unsigned int d1 = 0; d1 < dim; ++d1)
+    {
+      for(unsigned int d2 = 0; d2 < dim; ++d2)
+      {
+        for(unsigned int v = 0; v < dealii::VectorizedArray<Number>::size(); ++v)
+        {
+          dealii::Point<dim> q_point;
+
+          for(unsigned int i = 0; i < dim; ++i)
+            q_point[i] = q_points[i][v];
+
+          function->set_time(time);
+
+          // Assuming row-major vector function equivalent
+          value[d1][d2][v] = function->value(q_point, d1 * dim + d2);
```

Comment:
`dealii::Tensor` provides these functions as `component_to_unrolled_index()`  and `unrolled_to_component_indices()`. I wanted to be explicit here precisely because this determines how the function must be defined in the application. I guess we can provide ExaDG wrappers for the dealii functions for use in applications. How would you like to go on with this, also considering the conversation below?

Reply:
What about using `component_to_unrolled_index()` in the present PR? The other function or ExaDG wrappers for these functions could be introduced now or when needed (given that the present PR does not include examples showing the use of the new functionality). In essense, I want to get rid of the comment "assuming row-major ...". Using the respective functions, the assumption behind should be an implementation detail not relevant for ExaDG, right?

Reply:
Totally fine using  `component_to_unrolled_index()` here. I will make sure to consider this when I have concrete applications using this or have written unit tests. I am planning to do that very soon. I agree that this is a better solution for now.